python/test/unit/language/test_core.py

```diff
@@ -380,28 +387,36 @@ def get_scalar(x, dtype, low, high, filter):
         do_test(x, y[:1].reshape(()), kernel_broadcast_rhs)
 
 
-def _mod_operation_ill_conditioned(dtype_x, dtype_y) -> bool:
-    # FIXME For large x, we are casting x to a floating point where it does not fit
-    #       For small y, we are computing floor(div(float(x), y)) which may not fit
-    return (dtype_x, dtype_y) in [
-        ('int32', 'bfloat16'),
-        ('int32', 'float16'),
-        ('int32', 'float32'),
-        ('int64', 'bfloat16'),
-        ('int64', 'float16'),
-        ('int64', 'float32'),
-        ('int64', 'float64'),
-        ('uint16', 'bfloat16'),
-        ('uint16', 'float16'),
-        ('uint16', 'float32'),
-        ('uint32', 'bfloat16'),
-        ('uint32', 'float16'),
-        ('uint32', 'float32'),
-        ('uint64', 'bfloat16'),
-        ('uint64', 'float16'),
-        ('uint64', 'float32'),
-        ('uint64', 'float64'),
-    ]
+def _min_max_integral_mod_value(dtype_x, dtype_y) -> Optional[int]:
+    """
+    Limit min/max values for integral types for mod values. Leads to
+    overflow/underflow when casting large integral types to floats.
+    """
+    x_bitwidth = _bitwidth(dtype_x)
+    y_bitwidth = _bitwidth(dtype_y)
+
+    # uint types have larger range, must limit min/max values
+    if x_bitwidth < y_bitwidth and dtype_x in int_dtypes:
+        return None, None
```

Comment:
there's something still funny going on for some pairs of dtypes. 
For example here, for `int32` and `fp64`, when the rhs is a scalar, it's treated as just a float. Then that forces `int32` to be casted to  `fp32`, where it doesn't fit.

Not sure if all the errors in CI are of this shape, but yeah.

Reply:
Simplified the logic a bit and hard-capped the max value bit-width to be 32 if both types are 64 bit-width. Modulo operator is cursed...